mule-extensions-api/src/main/java/org/mule/runtime/extension/api/client/ExtensionsClient.java

```diff
@@ -0,0 +1,86 @@
+ * need of much manual coding and allowing to scale quickly all of this without a the need of compile dependency to
+ * the executed extension.
+ * <p>
+ * This API is simple and easy to use, the user just needs to define the extension and the corresponding operation
+ * that want's to invoke and pass a set of named parameters that will be attached when the operation gets executed. This implies
+ * that the user should know the parameter names and types to create them using an {@link OperationParameters} instance. The idea
+ * is that different extensions can provide different {@link OperationParameters} implementations for key operations.
+ * <p>
+ * This client lets the user reference complex configurations defined in the application and also has the capability to resolve
+ * expressions.
+ * <p>
+ * Note that this client will be reachable through the mule registry and you will be able to inject it in any class with lifecycle.
+ * <p>
+ * An usage example for an operation with this signature {@code public String getName(@UseConfig config, int account)} could be:
+ * <pre>
+ * {@code
+ * public class UsingExtensionsClient {
+ *  @Inject ExtensionsClient client;
+ *  ...
+ *  public void executeWithClient() {
+ *    OperationParameters parameters = DefaultOperationParameters.builder().configName("conf").addParameter("account", 12).build();
+ *    Result<String, Attributes> result = client.execute(CustomExtension.class, "getName", parameters);
+ *    if (result.getOutput().equals("DeveloperAccount"))
+ *    ...
+ *  }
+ * }
+ * }
+ * </pre>
+ *
+ * @since 1.0
+ */
+public interface ExtensionsClient {
+
+  /**
+   * Executes an operation asynchronously by returning a {@link Publisher} instance through which the user can obtain the
+   * result without blocking.
+   * <p>
+   * By subscribing to the returned {@link Publisher} the user can cover all the cases of processing, error and completion of the
+   * data stream.
+   * <p>
+   * When the executed operation is non-blocking, this is the recommended method to use
+   * <p>
+   * If the executed operation is not asynchronous in nature, the client might choose to actually execute in a synchronous manner.
+   *
+   * @param extension  the name of the extension that contains the operation to be executed.
+   * @param operation  the name of the operation to be executed.
+   * @param parameters an {@link OperationParameters} instance with all the parameters required to execute the operation.
+   * @return a {@link Publisher} of {@link Result}s with the payload content and the corresponding attributes to work with
+   * the operation execution asynchronously.
+   * @throws MuleException in any error case except for operation execution errors, those are handled with the {@link Publisher}.
+   */
+  <T, A extends Attributes> Publisher<Result<T, A>> executeAsync(String extension, String operation,
```

Comment:
The fact that it will be in JDK9 when it's released doesn't mean much IMO when it's not really usable without a framework. Do we really expect users to create subscribers which create subscriptions and do request(1) and all that just to use it, I don't think that's reasonable really?  These is also the fact that semantically it's not a great fit, for what should be a fairly tight API.

Can you explain more clearly what your concern with `CompletaleFuture`? It is not for dispatching the task, not sure what you are referring to there. It has `xxAsync()` methods yes, but this simply allows permit async operations to be performed after completion.  These aren't required, but don't _need_ to be used and there are advantages to using standard Java rather than defining our own interface e.g. it makes it easier to integrate with libs, including RxJava1.